### PR TITLE
Pull Request for Issue1814: Cleaning up pri files

### PR DESCRIPTION
**Pull request by iainworkman**
- Added AMBuildSettings.pri, which includes all the locations of the include/lib directories for the compositeCommon pri files.
- Added the ability to create a user customized AMBuildConfig.pri file, which over writes the default setting specified in AMBuildSettings.pri
- Altered qmake.cache.in to ensure the AMBuildSettings.pri file is loaded when building.
- Altered the composite common files to reference the settings specified in AMBuildSettings.pri
- Altered the composite common files to perform checks at the qmake stage for the existence of the required include/lib folders.
- Added AMBuildConfig_xxxtemplate.pri files for common build environments (Mac and Jenkins)
- Altered SGM Scan Controllers which made use of intptr_t to explicitly include <stdint.h> (required to build on more modern G++ versions)

### CODE REVIEW

**Comment by iainworkman** [comment]
Hey @XuetaoLiu-CLS, would you mind having a look over this one?
Thanks.

File: AMBuildSettings.pri [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+MOBILITY_QT_LIB_DIR = 
+
+# Location of the MPlot header and library files
+MPLOT_INCLUDE_DIR = $$PATH_TO_AM/../MPlot/include
+MPLOT_LIB_DIR = $$PATH_TO_AM/../MPlot/lib
+
+# location of the MPlot header and library files
+EIGEN_INCLUDE_DIR = $$PATH_TO_AM/source/Eigen
+
+# location of the MPlot header and source files
```
Comment (XuetaoLiu-CLS) [inline AMBuildSettings.pri]: update comments. It should be **AMDS **here

File: AMBuildSettings.pri [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# Whether to attempt to use Qt mobility, and the location of its header and library files
+USE_MOBILITY_QT = 0
+MOBILITY_QT_INCLUDE_DIR = 
+MOBILITY_QT_LIB_DIR = 
+
+# Location of the MPlot header and library files
+MPLOT_INCLUDE_DIR = $$PATH_TO_AM/../MPlot/include
+MPLOT_LIB_DIR = $$PATH_TO_AM/../MPlot/lib
+
+# location of the MPlot header and library files
```
Comment (XuetaoLiu-CLS) [inline AMBuildSettings.pri]: And this should be `Eigen` instead of `MPlot`

**Comment by XuetaoLiu-CLS** [comment]
Looks good other than some copy/paste comments issues.

Nice work! :100: 

Time Estimate: 15 minutes

**Comment by dretrex** [comment]
Looks good.
Time Estimate: 5 minutes

**Pull request merged**